### ISSUE TITLE
Change event is fired using Reactive form

### ISSUE THREAD

**Issue by jiangyh1024** [chymz/ng2-ckeditor#256]
```
    <ckeditor (change)="onDetailDataChanged()" formControlName="guidance" class="ckeditor" [config]="ckeConfig"
          debounce="500">
        </ckeditor>

  public onDetailDataChanged(): void {
    console.log(this.formGroup.getRawValue(), this.formGroup.dirty);
  }

// res.data is loaded from server side
              this.formGroup.patchValue({
                ...res.data,
                lookupTypeId
              }, { emitEvent: false });
```

after calling patch value, the change event is fired? Why ?

How to solve the problem?

**Comment by kzimny** [comment]
Do you change the value of guidance formControl with .patchValue?

**Comment by jiangyh1024** [comment]
> Do you change the value of guidance formControl with .patchValue?

Yes. And I found that the  (change) event of ckeditor triggered after I input some text instead of triggered after losing focus like <input (change)="changed($event)" /> does

**Comment by kzimny** [comment]
This is an expected behaviour. Change event trigger after the value has change. With debounce="500" you can add a delay when updating your value. If you wish to trigger an event after the focus change you use (blur)="onBlur($event)" instead of (change)="onDetailDataChanged()" event.

**Comment by jiangyh1024** [comment]
> This is an expected behaviour. Change event trigger after the value has change. With debounce="500" you can add a delay when updating your value. If you wish to trigger an event after the focus change you use (blur)="onBlur($event)" instead of (change)="onDetailDataChanged()" event.

Thanks for replying. I've decided to use onBlur event!
